test/test_torch.py

```diff
@@ -4629,24 +4629,66 @@ def test(name):
     def test_signal_window_functions(self):
         self._test_signal_window_functions(self)
 
-    @skipIfNoLapack
-    def test_inverse(self):
-        M = torch.randn(5, 5)
+    @staticmethod
+    def _test_inverse(self, cast):
+        from common import random_fullrank_matrix_distinct_singular_value
+
+        # no batches: 2-D tensors
+        M = cast(random_fullrank_matrix_distinct_singular_value(5))
         MI = torch.inverse(M)
-        E = torch.eye(5)
-        self.assertFalse(MI.is_contiguous(), 'MI is contiguous')
+        E = cast(torch.eye(5))
         self.assertEqual(E, torch.mm(M, MI), 1e-8, 'inverse value')
         self.assertEqual(E, torch.mm(MI, M), 1e-8, 'inverse value')
 
-        MII = torch.Tensor(5, 5)
+        MII = cast(torch.Tensor(5, 5))
         torch.inverse(M, out=MII)
-        self.assertFalse(MII.is_contiguous(), 'MII is contiguous')
         self.assertEqual(MII, MI, 0, 'inverse value in-place')
         # second call, now that MII is transposed
         torch.inverse(M, out=MII)
-        self.assertFalse(MII.is_contiguous(), 'MII is contiguous')
         self.assertEqual(MII, MI, 0, 'inverse value in-place')
 
+        # one batch
+        M = cast(random_fullrank_matrix_distinct_singular_value(5, 1))
+        MI = torch.inverse(M)
+        expected_inv = M.squeeze(0).inverse()
+        self.assertEqual(MI, expected_inv.unsqueeze(0))
+
+        # four batches
+        M = cast(random_fullrank_matrix_distinct_singular_value(5, 4))
+        expected_inv_list = []
+        for i in range(0, 4):
+            expected_inv_list.append(torch.inverse(M[i]))
+        expected_inv = torch.stack(expected_inv_list)
+        MI = torch.inverse(M)
+        self.assertEqual(MI, expected_inv)
+
+        # correctness test
+        M = cast(random_fullrank_matrix_distinct_singular_value(5, 3))
+        MI = torch.inverse(M)
+        self.assertEqual(torch.matmul(M, MI), E.expand_as(M))
+        self.assertEqual(torch.matmul(MI, M), E.expand_as(M))
+
+        # torch.inverse with out and batches
+        M = cast(random_fullrank_matrix_distinct_singular_value(5, 3))
+        MI = torch.Tensor(3, 5, 5)
+        torch.inverse(M, out=MI)
+        self.assertEqual(torch.inverse(M), MI)
+
```

Comment:
We only test `torch.inverse` on tensors of the form `(B, N, N)` and `(N, N)`. Would it be possible to add one test for a tensor of the form `(B1, ..., BK, N, N)`?